planning/behavior_path_start_planner_module/src/start_planner_module.cpp

```diff
@@ -616,43 +616,62 @@ PriorityOrder StartPlannerModule::determinePriorityOrder(
 }
 
 bool StartPlannerModule::findPullOutPath(
-  const std::vector<Pose> & start_pose_candidates, const size_t index,
-  const std::shared_ptr<PullOutPlannerBase> & planner, const Pose & refined_start_pose,
-  const Pose & goal_pose)
+  const Pose & start_pose_candidate, const std::shared_ptr<PullOutPlannerBase> & planner,
+  const Pose & refined_start_pose, const Pose & goal_pose)
 {
-  // Ensure the index is within the bounds of the start_pose_candidates vector
-  if (index >= start_pose_candidates.size()) return false;
+  const auto & dynamic_objects = planner_data_->dynamic_object;
+  const auto pull_out_lanes = start_planner_utils::getPullOutLanes(
+    planner_data_, planner_data_->parameters.backward_path_length + parameters_->max_back_distance);
+  const auto & vehicle_footprint = createVehicleFootprint(vehicle_info_);
+  // extract stop objects in pull out lane for collision check
+  const auto [pull_out_lane_objects, others] =
+    utils::path_safety_checker::separateObjectsByLanelets(
+      *dynamic_objects, pull_out_lanes, utils::path_safety_checker::isPolygonOverlapLanelet);
+  const auto pull_out_lane_stop_objects = utils::path_safety_checker::filterObjectsByVelocity(
+    pull_out_lane_objects, parameters_->th_moving_object_velocity);
```

Comment:
doning filterObjectsByVelocity first may be faster

Reply:
agree! thanks!

Reply:
changed in https://github.com/autowarefoundation/autoware.universe/pull/5952/commits/e5717adb0d8235d7393240ef3b3fbd53150c48bb